File: Regression/Zookeeper/Partner_Disabled/Partner_Disabled.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
from selenium.webdriver.common.action_chains import ActionChains

import time
import xlrd
import os

if os.environ.get('USE_PHANTOM'):
    print("using PhantomJS")

    driver = webdriver.PhantomJS()
    driver.implicitly_wait(5)
    driver.set_window_size(1120, 550)
else:
    print("using Firefox")

driver = webdriver.Firefox()
driver.get("http://www.pt.energypluscompany.com/newadmin/login.php")

elem = driver.find_element_by_name("loginusername").send_keys("mpeters")
elem = driver.find_element_by_name("loginpassword").send_keys("energy")
elem = driver.find_element_by_class_name("btn").click()
time.sleep(2)

## Disable Partner
driver.get("http://www.pt.energypluscompany.com/newadmin/partner_management.php")
time.sleep(2)
elem = driver.find_element_by_id("state_4").click() #PA
time.sleep(2)
elem = driver.find_element_by_id("pid_9440").click() #Virgin America
time.sleep(2)
elem = driver.find_element_by_id("partner_status_3").click() #disable
time.sleep(2)

N = 47
actions = ActionChains(driver) 
for _ in range(N):
    actions = actions.send_keys(Keys.TAB)
actions.perform()

elem = driver.find_element_by_id("savePartner").send_keys(Keys.ENTER);
time.sleep(3)
driver.close()

## Verify Partner Does Not Exist
driver = webdriver.Firefox()
driver.get("http://www.pt.energypluscompany.com/virginamerica/pa/")

if "The offer you are looking for is no longer available." in driver.page_source:
   print("Offer Expired - Passed")
else:
    print("Offer Expired - Failed")
time.sleep(2)

## Verify Partner Exists
driver.get("http://staging.devepc.com/combined/virginamerica/pa")

if "The offer you are looking for is no longer available." in driver.page_source:
    print ("Offer Expired - Passed")
else:
    print("Offer Expired - Failed")
driver.close()

# Enable Partner
driver = webdriver.Firefox()
driver.get("http://www.pt.energypluscompany.com/newadmin/login.php")

elem = driver.find_element_by_name("loginusername").send_keys("mpeters")
elem = driver.find_element_by_name("loginpassword").send_keys("energy")
elem = driver.find_element_by_class_name("btn").click()
time.sleep(2)

## Activate Partner
driver.get("http://www.pt.energypluscompany.com/newadmin/partner_management.php")
time.sleep(2)
elem = driver.find_element_by_id("state_4").click() #PA
time.sleep(2)
elem = driver.find_element_by_id("pid_9440").click() #Virgin America
time.sleep(2)
elem = driver.find_element_by_id("partner_status_1").click() #Active
time.sleep(2)

N = 47
actions = ActionChains(driver) 
for _ in range(N):
    actions = actions.send_keys(Keys.TAB)
actions.perform()

elem = driver.find_element_by_id("savePartner").send_keys(Keys.ENTER);
time.sleep(3)
driver.close()

File: Regression/Zookeeper/Partner_Disabled/Partner_Disabled.rb
require 'rubygems'
require 'watir-webdriver'
require 'win32ole' 
require 'csv'

browser = Watir::Browser.new :firefox
browser.goto "http://www.pt.energypluscompany.com/newadmin/login.php"

browser.text_field(:name => 'loginusername').set("mpeters")
browser.text_field(:name => 'loginpassword').set("energy")
browser.button(:value => 'Login').click
sleep 2

browser.link(:xpath => '/html/body/div[4]/div[2]/div[2]/ul/li[2]/a').click
sleep 2
browser.button(:value => 'PA').click
sleep 2
browser.table.td(:id => 'pid_9440').click
sleep 2
browser.button(:value => 'Disabled').click
sleep 2

wsh = WIN32OLE.new('Wscript.Shell')
wsh.AppActivate('Partners')
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :enter
sleep 5

browser.close

browser = Watir::Browser.new :firefox

browser.goto 'http://www.pt.energypluscompany.com/virginamerica/pa/'
if
browser.text.include? "The offer you are looking for is no longer available."
print "Offer Expired - Passed", "\n"
else 
print "Offer Expired - Failed", "\n"
end
sleep 2
browser.goto 'http://staging.devepc.com/combined/virginamerica/pa'
if
browser.text.include? "The offer you are looking for is no longer available."
print "Offer Expired - Passed"
else 
print "Offer Expired - Failed"
end

browser.close

browser = Watir::Browser.new :firefox
browser.goto "http://www.pt.energypluscompany.com/newadmin/login.php"

browser.text_field(:name => 'loginusername').set("mpeters")
browser.text_field(:name => 'loginpassword').set("energy")
browser.button(:value => 'Login').click
sleep 4

browser.link(:xpath => '/html/body/div[4]/div[2]/div[2]/ul/li[2]/a').click
sleep 5
browser.button(:value => 'PA').click
sleep 2
browser.table.td(:id => 'pid_9440').click
sleep 2
browser.button(:value => 'Active').click
sleep 5

wsh = WIN32OLE.new('Wscript.Shell')
wsh.AppActivate('Partners')
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :tab
browser.send_keys :enter
sleep 1

browser.close